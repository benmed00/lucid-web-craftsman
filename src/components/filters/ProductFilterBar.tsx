import React from 'react';
import { Badge } from '@/components/ui/badge';

interface ProductFilterBarProps {
  activeFilter: string;
  onFilterChange: (filterCategory: string) => void;
  currentSort: string; // Example: 'popular', 'price-asc', 'price-desc', 'newest'
  onSortChange: (sortKey: string) => void;
  // We can add more props later, like available categories if they become dynamic
}

const ProductFilterBar: React.FC<ProductFilterBarProps> = ({
  activeFilter,
  onFilterChange,
  currentSort,
  onSortChange,
}) => {
  // Define categories - could be dynamic in a real app
  const categories = [
    { key: 'all', label: 'Tous les produits' },
    { key: 'sacs', label: 'Sacs' },
    { key: 'chapeaux', label: 'Chapeaux' },
    // Add more categories as needed
  ];

  const sortOptions = [
    { key: 'popular', label: 'Trier par: Populaire' },
    { key: 'price-asc', label: 'Prix: Croissant' },
    { key: 'price-desc', label: 'Prix: Décroissant' },
    { key: 'newest', label: 'Nouveautés' },
  ];

  return (
    <div className="container mx-auto px-4 mb-12">
      <div className="flex flex-col md:flex-row justify-between items-start md:items-center gap-4 mb-8">
        <div className="flex flex-wrap gap-2">
          {categories.map((category) => (
            <Badge
              key={category.key}
              variant="outline"
              className={`cursor-pointer ${
                activeFilter === category.key
                  ? "border-olive-300 bg-olive-50 text-olive-800"
                  : "border-stone-300 hover:border-olive-300 hover:bg-olive-50 hover:text-olive-800"
              }`}
              onClick={() => onFilterChange(category.key)}
            >
              {category.label}
            </Badge>
          ))}
        </div>

        <div className="flex gap-2">
          <select 
            className="text-sm border border-stone-300 rounded-md py-2 px-3 focus:outline-none focus:border-olive-400"
            value={currentSort}
            onChange={(e) => onSortChange(e.target.value)}
          >
            {sortOptions.map(option => (
              <option key={option.key} value={option.key}>{option.label}</option>
            ))}
          </select>
        </div>
      </div>
    </div>
  );
};

export default ProductFilterBar;
