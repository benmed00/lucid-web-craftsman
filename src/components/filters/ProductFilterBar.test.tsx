// src/components/filters/ProductFilterBar.test.tsx
import { describe, it, expect, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import ProductFilterBar from './ProductFilterBar'; // Assuming path is correct

describe('ProductFilterBar', () => {
  const mockOnFilterChange = vi.fn();
  const mockOnSortChange = vi.fn();

  beforeEach(() => {
    mockOnFilterChange.mockClear();
    mockOnSortChange.mockClear();
  });

  it('renders filter badges and sort dropdown', () => {
    render(
      <ProductFilterBar
        activeFilter="all"
        onFilterChange={mockOnFilterChange}
        currentSort="popular"
        onSortChange={mockOnSortChange}
      />
    );

    expect(screen.getByText('Tous les produits')).toBeInTheDocument();
    expect(screen.getByText('Sacs')).toBeInTheDocument();
    expect(screen.getByText('Chapeaux')).toBeInTheDocument();
    expect(screen.getByDisplayValue('Trier par: Populaire')).toBeInTheDocument(); // Check by label
  });

  it('highlights the active filter badge', () => {
    render(
      <ProductFilterBar
        activeFilter="sacs"
        onFilterChange={mockOnFilterChange}
        currentSort="popular"
        onSortChange={mockOnSortChange}
      />
    );
    // Active badge should have specific classes (this depends on implementation)
    // Example: check for classes that include 'bg-olive-50' and 'text-olive-800'
    const activeBadge = screen.getByText('Sacs');
    expect(activeBadge).toHaveClass('border-olive-300'); // More specific check if possible
    expect(activeBadge).toHaveClass('bg-olive-50');
    expect(activeBadge).toHaveClass('text-olive-800');

    const inactiveBadge = screen.getByText('Tous les produits');
    expect(inactiveBadge).not.toHaveClass('bg-olive-50');
  });

  it('calls onFilterChange with the correct category when a filter badge is clicked', async () => {
    const user = userEvent.setup();
    render(
      <ProductFilterBar
        activeFilter="all"
        onFilterChange={mockOnFilterChange}
        currentSort="popular"
        onSortChange={mockOnSortChange}
      />
    );

    const sacsBadge = screen.getByText('Sacs');
    await user.click(sacsBadge);

    expect(mockOnFilterChange).toHaveBeenCalledTimes(1);
    expect(mockOnFilterChange).toHaveBeenCalledWith('sacs');
  });

  it('calls onSortChange with the correct sort key when a sort option is selected', async () => {
    const user = userEvent.setup();
    render(
      <ProductFilterBar
        activeFilter="all"
        onFilterChange={mockOnFilterChange}
        currentSort="popular" // Initial sort value
        onSortChange={mockOnSortChange}
      />
    );

    const sortDropdown = screen.getByRole('combobox');
    // Select by the value of the option
    await user.selectOptions(sortDropdown, 'price-asc');

    expect(mockOnSortChange).toHaveBeenCalledTimes(1);
    expect(mockOnSortChange).toHaveBeenCalledWith('price-asc');
  });

  it('displays the currentSort value in the dropdown', () => {
    render(
      <ProductFilterBar
        activeFilter="all"
        onFilterChange={mockOnFilterChange}
        currentSort="price-desc" // Set a specific sort
        onSortChange={mockOnSortChange}
      />
    );
    // The <select> element's value will be the key, e.g., 'price-desc'
    // The displayed text for this option will be 'Prix: DÃ©croissant'
    const selectElement = screen.getByRole('combobox') as HTMLSelectElement;
    expect(selectElement.value).toBe('price-desc');

    // To check the displayed label of the selected option:
    const selectedOption = Array.from(selectElement.options).find(opt => opt.selected);
    expect(selectedOption?.textContent).toBe('Prix: DÃ©croissant');
  });

  it('matches snapshot', () => {
    const { container } = render(
      <ProductFilterBar
        activeFilter="all"
        onFilterChange={mockOnFilterChange}
        currentSort="popular"
        onSortChange={mockOnSortChange}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
