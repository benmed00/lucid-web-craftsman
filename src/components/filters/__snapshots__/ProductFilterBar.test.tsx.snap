// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductFilterBar > matches snapshot 1`] = `
<div>
  <div
    class="container mx-auto px-4 mb-12"
  >
    <div
      class="flex flex-col md:flex-row justify-between items-start md:items-center gap-4 mb-8"
    >
      <div
        class="flex flex-wrap gap-2"
      >
        <div
          class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 cursor-pointer border-olive-300 bg-olive-50 text-olive-800"
        >
          Tous les produits
        </div>
        <div
          class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-foreground cursor-pointer border-stone-300 hover:border-olive-300 hover:bg-olive-50 hover:text-olive-800"
        >
          Sacs
        </div>
        <div
          class="inline-flex items-center rounded-full border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 text-foreground cursor-pointer border-stone-300 hover:border-olive-300 hover:bg-olive-50 hover:text-olive-800"
        >
          Chapeaux
        </div>
      </div>
      <div
        class="flex gap-2"
      >
        <select
          class="text-sm border border-stone-300 rounded-md py-2 px-3 focus:outline-none focus:border-olive-400"
        >
          <option
            value="popular"
          >
            Trier par: Populaire
          </option>
          <option
            value="price-asc"
          >
            Prix: Croissant
          </option>
          <option
            value="price-desc"
          >
            Prix: Décroissant
          </option>
          <option
            value="newest"
          >
            Nouveautés
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;
